api/app/src/providers/fitbit.ts

```diff
@@ -332,4 +336,67 @@ export class Fitbit extends Provider implements OAuth2 {
       }
     );
   }
+
+  async postAuth(token: string, userId?: string) {
+    const accessToken = JSON.parse(token).access_token;
+
+    const scopes = JSON.parse(token).scope;
+
+    const subscriptionTypes: Record<FitbitScopes, string> = {
+      [FitbitScopes.activity]: "activities",
+      [FitbitScopes.nutrition]: "foods",
+      [FitbitScopes.weight]: "body",
+      [FitbitScopes.sleep]: "sleep",
+    };
+
+    const subscriptionId = userId;
+
+    if (this.allRequiredScopesIncluded(scopes)) {
+      const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/apiSubscriptions/${subscriptionId}.json`;
+      this.createSubscription(subscriptionUrl, accessToken);
+      console.log("All scopes included. User subscribed to all Fitbit WH collectionTypes.");
+    } else {
+      for (const scope of Object.keys(subscriptionTypes)) {
+        if (scopes.includes(scope)) {
+          const subscriptionType = subscriptionTypes[scope as FitbitScopes];
```

Comment:
Defs NABD, but we can use `Object.entries` and avoid the type override with `as` - [example](https://github.com/metriport/metriport/blob/c77eacae50fdd3fcf56adb64af186ba914b0ff48/packages/infra/lib/shared/secrets.ts#L13).